drivers/led/is31fl3194/is31fl3194.c

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright (c) 2022 Benjamin Björnsson <benjamin.bjornsson@gmail.com>.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT issi_is31fl3194
+
+/**
+ * @file
+ * @brief LED driver for the IS31FL3194 I2C LED driver
+ */
+
+#include <zephyr/zephyr.h>
+#include <zephyr/drivers/led.h>
+#include <zephyr/drivers/i2c.h>
+#include <string.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(is31fl3194, CONFIG_LED_LOG_LEVEL);
+
+#include "is31fl3194_reg.h"
+
+#define IS31FL3194_INDEX_MAX 2
+
+struct is31fl3194_config {
+	struct is31fl3194_ctx ctx;
+	const struct i2c_dt_spec i2c;
+	uint8_t num_leds;
+	const struct led_info *leds_info;
+	const uint8_t *curr_map_max;
+};
+
+struct is31fl3194_data {
+	struct is31fl3194_regs regs;
+};
+
+int is31fl3194_i2c_read(const struct i2c_dt_spec *i2c_spec, uint8_t reg_addr, uint8_t *value,
+			uint8_t len)
+{
+	return i2c_burst_read_dt(i2c_spec, reg_addr, value, len);
+}
+
+int is31fl3194_i2c_write(const struct i2c_dt_spec *i2c_spec, uint8_t reg_addr, uint8_t *value,
+			 uint8_t len)
+{
+	return i2c_burst_write_dt(i2c_spec, reg_addr, value, len);
+}
+
+int is31fl3194_on(const struct device *dev, uint32_t led)
+{
+	const struct is31fl3194_config *config = dev->config;
+	struct is31fl3194_ctx *ctx = (struct is31fl3194_ctx *)&config->ctx;
+	int res;
+
+	res = is31fl3194_out_en_set(ctx, led, 1);
```

Comment:
Agree, it currently just enables it. Does turning it on mean that it should be set to full brightness?

Reply:
Since it is not specified, it is up to you. But I think that full brightness should be right. Note that I am not a big fan of the `led_off` / `led_on` functions. They don't exist in the Linux LED API for example. I think of them pretty much as wrappers to `led_set_brightness` with brightness set to 0 or 100.